lib/string_reverse.rb

```diff
@@ -1,4 +1,28 @@
 # A method to reverse a string in place.
+require 'pry'
 def string_reverse(my_string)
-  my_string << "not implemented"
+  if my_string == nil || my_string.length == 0
+    return my_string
+  end
+
+  array = my_string.split("")

```

Comment:
Converting the string to an array using split is adding to your space complexity. Ping me on slack if you have further questions on this after thinking through this. 
Instead, try deleting line 8 and replace all instances of `array` that follow with `my_string`

Reply:
I'm curious are the tests passing for you? Since you are converting the string into and array and then reversing the array, the original string is not getting reversed.